src/Analysis/Codelyzer.Analysis.CSharp/CSharpRoslynProcessor.cs

```diff
@@ -76,152 +76,91 @@ public override UstNode VisitUsingDirective(UsingDirectiveSyntax node)
 
-            AddAttributeNodesToList(node, handler.UstNode.Children);
-            AddIdentifierNameNodesToList(node, handler.UstNode.Children);
-        
             return handler.UstNode;
         }
-
         public override UstNode VisitConstructorDeclaration(ConstructorDeclarationSyntax node)
         {
             ConstructorDeclarationHandler handler = new ConstructorDeclarationHandler(_context, node);
-            
-            AddMethodBodyNodesToList(node, handler.UstNode.Children);
-            AddIdentifierNameNodesToList(node, handler.UstNode.Children);
-
             return handler.UstNode;
         }
-
         public override UstNode VisitMethodDeclaration(MethodDeclarationSyntax node)
         {
             MethodDeclarationHandler handler = new MethodDeclarationHandler(_context, node);
-
-            AddMethodBodyNodesToList(node, handler.UstNode.Children);
-            AddIdentifierNameNodesToList(node, handler.UstNode.Children);
-
             return handler.UstNode;
         }
-
         public override UstNode VisitReturnStatement(ReturnStatementSyntax node)
         {
             ReturnStatementHandler handler = new ReturnStatementHandler(_context, node);
-            AddInvocationExpressionNodesToList(node, handler.UstNode.Children);
-            AddObjectCreationNodesToList(node, handler.UstNode.Children);
-            AddLiteralExpressionNodesToList(node, handler.UstNode.Children);
-            AddIdentifierNameNodesToList(node, handler.UstNode.Children);
-
             return handler.UstNode;
         }
-
         public override UstNode VisitBlock(BlockSyntax node)
         {
             BlockStatementHandler handler = new BlockStatementHandler(_context, node);
-            var result = handler.UstNode;
-
-            AddInvocationExpressionNodesToList(node, result.Children);
-            AddReturnStatementNodesToList(node, result.Children);
-            AddObjectCreationNodesToList(node, result.Children);
-            AddLiteralExpressionNodesToList(node, result.Children);
-            AddIdentifierNameNodesToList(node, result.Children);
-
-            return result;
+            if (!string.IsNullOrEmpty(handler.UstNode.Identifier))
+            {
+                return handler.UstNode;
+            }
+            else
+            {
+                return null;
+            }
         }
-
-
         public override UstNode VisitArrowExpressionClause(ArrowExpressionClauseSyntax node)
         {
             ArrowExpressionClauseHandler handler = new ArrowExpressionClauseHandler(_context, node);
             var result = handler.UstNode;
 
-            AddInvocationExpressionNodesToList(node, result.Children);
-            AddObjectCreationNodesToList(node, result.Children);
-            AddLiteralExpressionNodesToList(node, result.Children);
-            AddIdentifierNameNodesToList(node, result.Children);
-
             return result;
```

Comment:
nit: instead of creating `var result`, we can probably just have `return handler.UstNode`. This change would also follow the return statement pattern of all other `Visit*()` methods.